### PR TITLE
fix: Add old datamover fields back to DPA

### PR DESCRIPTION
**Pull request by mateusoliveira43**
## Description

Revert removal of fields from DPA from https://github.com/openshift/oadp-operator/pull/1232, following the project deprecation process (only remove fields when API version updates). Still not allow users to use those fields.

## How to test

Create a DPA with vsm plugin or old datamover feature field. DPA should not reconcile and tell user to use Velero built-in datamover.

### CODE REVIEW

File: controllers/validator.go [hunk 1]
```diff
@@ -78,6 +78,16 @@ func (r *DPAReconciler) ValidateDataProtectionCR(log logr.Logger) (bool, error)
 		}
 	}
 
+	// check for VSM/Volsync DataMover (OADP 1.2 or below) syntax
+	for _, plugin := range dpa.Spec.Configuration.Velero.DefaultPlugins {
+		if plugin == oadpv1alpha1.DefaultPluginVSM {
+			return false, errors.New("Delete vsm from spec.configuration.velero.defaultPlugins and dataMover object from spec.features. Use Velero Built-in Data Mover instead")
+		}
+	}
```
Comment (kaovilai) [inline controllers/validator.go:86]: Move this into [`func (r *DPAReconciler) ValidateVeleroPlugins(log logr.Logger) (bool, error) {`](https://github.com/openshift/oadp-operator/blob/db116623f3e66ad9763514d562b688624cc44d00/controllers/validator.go#L121) ?

Comment (kaovilai) [reply controllers/validator.go:86]: there's already a for loop there.

Comment (mateusoliveira43) [reply controllers/validator.go:86]: going to check that out

**Comment by mateusoliveira43** [comment]
/remove-lifecycle stale
Need this in 1.4, otherwise upgrades may break

File: controllers/validator.go (hunk 1 above)
Comment (mateusoliveira43) [reply controllers/validator.go]: done

**Review by kaovilai — APPROVED** [review]

**Pull request merged**